internal/scriptlet/addrule.go

```diff
@@ -0,0 +1,147 @@
+package scriptlet
+
+import (
+	"errors"
+	"fmt"
+	"net"
+	"regexp"
+	"strings"
+)
+
+var (
+	// reAdguardScriptlet detects and extracts key data from AdGuard-style scriptlets.
+	reAdguardScriptlet = regexp.MustCompile(`(.*)#%#\/\/scriptlet\((.+)\)`)
+	// adguardToCanonical maps AdGuard scriptlet names to their respective implementations inside the scriptlet bundle.
+	adguardToCanonical = map[string]string{
+		"set-local-storage-item":   "setLocalStorageItem",
+		"set-session-storage-item": "setSessionStorageItem",
+		"nowebrtc":                 "nowebrtc",
+		"prevent-fetch":            "preventFetch",
+		"prevent-xhr":              "preventXHR",
+		"set-constant":             "setConstant",
+	}
+	// reUblockScriptlet detects and extracts key data from uBlock Origin-style scriptlets.
+	reUblockScriptlet = regexp.MustCompile(`(.*)##\+js\((.+)\)`)
+	// ublockToCanonical maps uBlock Origin scriptlet names to their respective implementations inside the scriptlet bundle.
+	ublockToCanonical = map[string]string{
+		// TODO: manually check ublock syntax compatibility
+		"set-local-storage-item": "setLocalStorageItem",
+		"no-xhr-if":              "preventXHR",
+		"no-fetch-if":            "preventFetch",
+		"nowebrtc":               "nowebrtc",
+		"set-constant":           "setConstant",
+	}
+	errNotQuotedString    = errors.New("not a quoted string")
+	errUnsupportedSyntax  = errors.New("unsupported syntax")
+	errEmptyScriptletBody = errors.New("scriptlet body is empty")
+)
+
+func (i *Injector) AddRule(rule string) error {
```

Comment:
`i` as the receiver name works here, but as the function grows, it could make the code harder to understand, especially since `i` is commonly used for loop counters. A more descriptive name, like `in` or `inj` or `injector`, might improve readability in the long run.

Reply:
Agreed.